src/main/java/com/traveltime/sdk/dto/requests/proto/Transportation.java

```diff
@@ -1,35 +1,193 @@
 package com.traveltime.sdk.dto.requests.proto;
 
+import com.igeolise.traveltime.rabbitmq.requests.RequestsCommon;
 import lombok.AllArgsConstructor;
+import lombok.Builder;
 import lombok.Getter;
 
 public interface Transportation {
-    String getValue();
-    Integer getCode();
+    TransportationType getType();
+    RequestsCommon.Transportation getProtoMessage();
 
+    @Builder
     @Getter
-    @AllArgsConstructor
-    enum Modes implements Transportation {
-        PUBLIC_TRANSPORT("pt", 0),
```

Comment:
We could avoid breaking source-code compatibility by introducing the default fields as:

```
  public static final Transportation PUBLIC_TRANSPORT = new PublicTransport()l
  public static final Transportation WALKING_FERRY = new WalkingFerry();
  ...
```
etc